src/main/java/org/usfirst/frc4048/common/swervev3/io/SparkMaxModuleIO.java

```diff
@@ -0,0 +1,87 @@
+package org.usfirst.frc4048.common.swervev3.io;
+
+import com.ctre.phoenix.sensors.WPI_CANCoder;
+import com.revrobotics.CANSparkMax;
+import org.usfirst.frc4048.common.Constants;
+import org.usfirst.frc4048.common.swervev2.KinematicsConversionConfig;
+import org.usfirst.frc4048.common.swervev2.SwerveIdConfig;
+
+import java.util.concurrent.atomic.AtomicLong;
+
+public class SparkMaxModuleIO implements ModuleIO {
+    private final CANSparkMax driveMotor;
+    private final CANSparkMax steerMotor;
+    private final WPI_CANCoder absEncoder;
+    private final AtomicLong steerOffset = new AtomicLong(0);
+
+    public SparkMaxModuleIO(SwerveIdConfig motorConfig, KinematicsConversionConfig conversionConfig, boolean driveInverted, boolean steerInverted) {
```

Comment:
I thought all the spark max related stuff should go into the SparkMaxModule, leaving the IO to handle logging

Reply:
this has been refactored